sir_complainsalot/delphi_sir_complainsalot/run.py

```diff
@@ -57,28 +59,64 @@ def report_complaints(all_complaints, params):
             # You will get a SlackApiError if "ok" is False
             assert False, e.response["error"]
 
-def format_complaints(complaints):
-    """Build a formatted Slack message for posting to the API.
-
-    To find good formatting for blocks, try the block kit builder:
-    https://api.slack.com/tools/block-kit-builder
-
-    """
 
+def get_maintainers_block(complaints):
     maintainers = set()
     for c in complaints:
         maintainers.update(c.maintainers)
-
-    blocks = [
-	{
+    
+    maintainers_block = {
 	    "type": "section",
 	    "text": {
 		"type": "mrkdwn",
 		"text": "Hi, this is Sir Complains-a-Lot. I need to speak to " +
                         (", ".join("<@{0}>".format(m) for m in maintainers)) + "."
 	    }
 	}
-    ]
+
+    return maintainers_block
+
+
+def format_complaints_aggregated_by_source(complaints):
+    """Build formatted Slack message for posting to the API, aggregating the 
+    complaints by source to reduce the number of blocks."""
+
+    blocks = [get_maintainers_block(complaints)]
+
+    def aggregated_message_for_source(x): return "{complaint} - (last update: {last_updated})".format(
+        complaint=x.message, last_updated=x.last_updated.strftime("%Y-%m-%d"))
```

Comment:
While we're not yet including linting of this package in the repo CI, probably best not to introduce *new* linting errors

Reply:
Eugh sorry -- I had pylint enabled in vscode but for some reason was not catching line length.